### PR TITLE
Unmerged fetch list

### PR DESCRIPTION
**Pull request by wzzju**
#### Support ParallelExecutor to fetch unmerged LoDTensors.

Add `return_merged` as a parameter in `Executor.run`. It indicates whether fetched variables (the variables specified in the fetch list) should be merged according to the execution device dimension. If `return_merged` is False, the type of the return value is a two-dimensional list of Tensor ( `return_numpy` is False) or a two-dimensional list of `numpy.ndarray` ( `return_numpy` is True). If `return_merged` is True, the type of the return value is an one-dimensional list of Tensor ( `return_numpy` is False) or an one-dimensional list of `numpy.ndarray` ( `return_numpy` is True). If the lengths of fetched results are variant, please set return_merged as False, which denotes that the fetched results will not be merged. The default is True, but it is just for the compatibility, and may use False as default value in the future version.

image
image
image

### CODE REVIEW

File: python/paddle/fluid/executor.py [hunk 1]
```diff
@@ -724,6 +725,9 @@ def run(self,
                 the input program is :code:`fluid.Program`, and the parameters(program, feed variable name
                 and fetch_list variable) of this interface remains unchanged during running.
                 The default is False.
+            return_merged(bool): This parameter indicates whether fetched variables (the variable specified
```
Comment (sneaxiy) [inline python/paddle/fluid/executor.py]: Add some examples to show the usage of this parameter?

Comment (lanxianghit) [inline python/paddle/fluid/executor.py]: >(the variable specified
(the variable**s** specified

File: python/paddle/fluid/executor.py [hunk 2]
```diff
@@ -724,6 +725,9 @@ def run(self,
...
+                in the fetch list) should be merged according to the execution device dimension. If it is
+                False, the fetched results will not be merged. The default is True.
```
Comment (lanxianghit) [inline python/paddle/fluid/executor.py]: 需要详细解释一下True和False情况下返回的数据结构，区别是什么，分别在什么情况下使用（目前来看应该是True只是为了兼容，后续都推荐使用False，并有可能在后续版本中改为默认False），最好给出带有实际数据形式的例子（可以添加在代码示例中或者单独说明）

Comment (lanxianghit) [reply python/paddle/fluid/executor.py]: 上述信息最好也在pr description中进行说明，以便于其他reviewer更快的明确功能点，另外，请 @qingqing01 @xyzhou-puck @xixiaoyao 看一下对接口和返回值设计是否有其他意见

File: python/paddle/fluid/executor.py (hunk 1 above)
Comment (wzzju) [reply python/paddle/fluid/executor.py]: Done.

Comment (wzzju) [reply python/paddle/fluid/executor.py]: Done.

File: python/paddle/fluid/executor.py (hunk 2 above)
Comment (wzzju) [reply python/paddle/fluid/executor.py]: Done.

**Review by sneaxiy — DISMISSED** [review]
LGTM.

**Review by xyzhou-puck — DISMISSED** [review]
LGTM.

**Review by qingqing01 — DISMISSED** [review]
LGTM.

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by Boyan-Liu — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]

**Pull request merged**